navigating-the-single-cell-landscape-a-comprehensive-workshop-on-scrnaseq/day-1/4-psuedobulk.qmd

```diff
@@ -17,208 +17,532 @@ editor_options:
+
+```{r}
+results(dds, contrast = c("condition", "Flu", "Normal"))
+```
+
+```{r}
+# run 'resultsNames(dds)' to see "name" options
 # current options: "condition_Flu_vs_Normal"  "condition_nCoV_vs_Normal"
 # this pulls results from the DESeq object
-res <- results(dds)
+res <- results(dds, name = "condition_Flu_vs_Normal")
 res
 
-# filtering and rearranging for plotting
-restbl <- res %>% data.frame %>% rownames_to_column('gene') %>% filter(pvalue < 0.05)
+# Summary function 
+summary(res)
+```
 
-# save the table!
-write.csv(restbl,file='./navigating-the-single-cell-landscape-a-comprehensive-workshop-on-scrnaseq/day-1/pseudobulk_analysis/results/cd4_annotation_pseudobulk_results.csv')
+Save Results\
+**NOTE: Save all data so if a collaborator asks what the comparison was, you have the header stored or rerun analysis.**
+
+```{r}
+# Uncomment below to create results folder
+# dir.create('pseudobulk_analysis/results')
+
+# Save as a R object
+saveRDS(object = res, 
+        file = 'pseudobulk_analysis/results/cd4_annot_pseudobulk_results.rds')
+
+# Filtering and rearranging for plotting
+restbl <- res %>% 
+  data.frame %>% 
+  rownames_to_column('gene')
+
+# Write data frame results to CSV
+write.csv(
+  restbl,
+  file='pseudobulk_analysis/results/cd4_annot_pseudobulk_results.csv'
+  )
 ```
 
+## Visualize Results
+
+Volcano plot is often the ultimate goal of differential expression analysis.
+
+### From CSV object
+
 ```{r}
-### DO THESE BEFORE RUNNING DESEQ()  - optional - this is from a tutorial
-# remove rows with less than 10 reads across all of the samples
-# keep <- rowSums(counts(dds)) >= 10
-# 
-# dds_1 <- dds[keep,]
-# dds_1
-# dds_1$condition <- relevel(dds_1$condition, ref = "Normal")
+res <- read_csv('pseudobulk_analysis/results/cd4_annot_pseudobulk_results.csv')
 ```
 
 ```{r}
-# dds_2 <- DESeq(dds_1)
-# res <- results(dds_2)
-# res
+library(EnhancedVolcano)
+EnhancedVolcano(res,
+                lab = res$gene,
+                x = 'log2FoldChange',
+                y = 'padj',
+                xlim = c(-8, 8),
+                ylim = c(0, 5),
+                title = 'Volcano Plot: CD4 T cells',
+                subtitle = 'Normal < -  - - -  | -  - - -  > Flu',
+                xlab = 'Log2 Fold Change',
+                ylab = '-log10 FDR',
+                pCutoff = 0.05,
+                FCcutoff = 1.5,
+                pointSize = c(ifelse(res$padj < 0.05 & abs(res$log2FoldChange) > 1.5, 1.5, 1)),
+                labSize = 4.0,
+                col = c('grey30', 'forestgreen', 'royalblue', 'red2'),
+                colAlpha = 1,
+                legendLabels=c('Not Significant', 'Log2FC', 'P-Value', 'Log2FC & P-Value'),
+                legendPosition = 'right',
+                legendLabSize = 14,
+                drawConnectors = TRUE,
+                widthConnectors = 0.5,
+                colConnectors = 'grey30')
+
 ```
 
+TD: List notable genes, highlight where they are found, what they do, etc
```

Comment:
I think what I would highlight is that there is a general shift towards many genes being turned ON in disease context.